### PR TITLE
feat: update default-binary to 7.0.12

### PR DESCRIPTION
**Pull request by orgads**
from 10.0

### CODE REVIEW

**Comment by hasezoey** [comment]
please explain what you are trying to do here as i do not understand. also have you read our [Mongodb Server Versions: When a Version gets upgraded guideline](https://typegoose.github.io/mongodb-memory-server/docs/guides/mongodb-server-versions#when-a-version-gets-upgraded)?

**Comment by orgads** [comment]
Mongo 6 doesn't support Debian bookworm. I just started preparing this for 10.x, but if you prefer to do it yourself later I'll close it.

I use `binary.version` in `create` for now.

**Comment by hasezoey** [comment]
> I just started preparing this for 10.x, but if you prefer to do it yourself later I'll close it.

10.x already has it upgraded in 2436d6300bd863efa3300103a3223d3849d0bda3 and has been included since `10.0.0-beta.1`

**Comment by orgads** [comment]
Oh, nice! I didn't notice the beta.

When do you plan to release?

**Comment by hasezoey** [comment]
> When do you plan to release?

see [10.0 milestone](https://github.com/typegoose/mongodb-memory-server/milestone/4).
TL;DR: if nothing goes wrong, on 18.07.2024
